sympy/solvers/ode/ode.py

```diff
@@ -731,6 +731,216 @@ def _helper_simplify(eq, hint, match, simplify=True, ics=None, **kwargs):
+    >>> eq = y(x).diff(x) - (x**2 + y(x)**2)/(x*y(x))
+
+    Transform the homogeneous equation to a linear equation
+
+    >>> forward_trans = {x:t, y(x): t*v(t)}
+    >>> reverse_trans = {t: x, v(t): y(x)/x}
+    >>> trans = (forward_trans, reverse_trans)
+    >>> homeq = dsubs(eq, trans).simplify()
+    >>> homeq
+    t*Derivative(v(t), t) - 1/v(t)
+
+    Find the solutions for the transformed equation
+
+    >>> solt = dsolve(homeq)
+    >>> solt
+    [Eq(v(t), -sqrt(C1 + 2*log(t))), Eq(v(t), sqrt(C1 + 2*log(t)))]
+
+    Re-substitute `v = y x` to get the solutions to the
+    original equation
+
+    >>> sol = list(map(lambda sol: dsubs(sol, trans[::-1]), solt))
+    >>> sol
+    [Eq(y(x)/x, -sqrt(C1 + 2*log(x))), Eq(y(x)/x, sqrt(C1 + 2*log(x)))]
+    >>> checkodesol(eq, sol)
+    [(True, 0), (True, 0)]
+
+    Let us look at another example - The 2nd order Cauchy Euler equation.
+
+    >>> x, a, b, t = symbols('x a b t')
+    >>> y, phi = symbols('y phi', cls=Function)
+    >>> eq = x**2*y(x).diff(x, 2) + a*x*y(x).diff(x) + b*y(x)
+
+    Transform the equation using the substitutions `x = e^t`
+    and `y(x) = \phi(t)`
+
+    >>> from sympy import exp, log, logcombine
+    >>> forward_trans = {x: exp(t), y(x): phi(t)}
+    >>> reverse_trans = {t: log(x), phi(t): y(x)}
+    >>> trans = (forward_trans, reverse_trans)
+    >>> eqtrans = dsubs(eq, trans).simplify()
+    >>> eqtrans
+    a*Derivative(phi(t), t) + b*phi(t) - Derivative(phi(t), t) + Derivative(phi(t), (t, 2))
+
+    The equation is now transformed to a constant-coefficient equation.
+
+    >>> solt = dsolve(eqtrans)
+    >>> solt
+    Eq(phi(t), C1*exp(t*(-a - sqrt(a**2 - 2*a - 4*b + 1) + 1)/2) + C2*exp(t*(-a + sqrt(a**2 - 2*a - 4*b + 1) + 1)/2))
+
+    Re-substitute for `\phi(t)` and `t` to get the solution for
+    the original equation.
+
+    >>> sol = logcombine(dsubs(solt, trans[::-1]), force=True)
+    >>> sol
+    Eq(y(x), C1*x**(-a/2 - sqrt(a**2 - 2*a - 4*b + 1)/2 + 1/2) + C2*x**(-a/2 + sqrt(a**2 - 2*a - 4*b + 1)/2 + 1/2))
+    >>> checkodesol(eq, sol)
+    (True, 0)
+    """
+    from sympy.core.function import Subs, Function, Derivative, AppliedUndef
+    from sympy.core.symbol import Symbol
+    from sympy.integrals.integrals import Integral
+
+    def diffx(e, sym, n):
+        if isinstance(sym, Function):
+            if n > 1:
+                return diffx(diffx(e, sym, n-1), sym, 1)
+            return Derivative(e, sym.args[0]) / Derivative(sym, sym.args[0])
+        else:
+            return Derivative(e, (sym, n))
+
+    trans, reverse_trans = trans
+    newvars = set(reverse_trans)
+    if not isinstance(reverse_trans, dict):
+        reverse_trans = None
+    old_funcs = eq.atoms(Function)
+    mapdict = {}
+
+    for var in trans:
+        if not isinstance(var, Symbol) and not isinstance(var, Function):
+            raise ValueError(f"Expected Symbol or Function in the transformation rule, recieved {var.__class__.__name__}")
```

Comment:
```suggestion
            raise ValueError(f"Expected Symbol or Function; received {var.__class__.__name__}")
```
Where this is occuring will be apparent when the error is raised.

Reply:
Sure, I'll change this in the next commit I push